### DISCUSSION TITLE
Figuring Out Zindex Between Sprites And ui Elements

### DISCUSSION THREAD

**Discussion by Magnitus-** [bevyengine/bevy#14816] [Q&A] [answered]
So following my last questions, where I tried to unsuccessfully apply rotation transforms to elements inside a ui menu (I almost got it to work by putting the rotated ImageBundle centered in a container, but then the cropping had some really weird behavior), I instead changed strategy and tried to play with the zindex and position of independent ui elements and sprite elements.

However, what I'm noticing now is that the ui element always appear on top of my sprite. In the ui element, I tried setting the zindex to low local or global values. In the sprite element, I tried putting the third parameter in the transition for the transform (as I understand it, the zindex for 2D) to different values including 1000000.0 and no matter what, the ui element is always on top.

Is that a deliberate decision? Or am doing something wrong and controlling the relative zindices of ui and non-ui component is an easily solvable problem?

**Comment by rparrett** [comment, marked as answer]
> Is that a deliberate decision

Yes, UI is drawn in a separate pass on top of other cameras. Messing with `Transform` of UI components is a sort of undefined behavior, and we don't have a way of doing arbitrary rotations of UI nodes.

It looks like you're doing a minimap? I would consider using a separate 2d camera, rendering it to a texture, and then displaying that texture in your UI with an `ImageBundle`. <https://bevyengine.org/examples/3d-rendering/render-to-texture/> should get you most of the way there.

**Reply by Magnitus-** [reply]
I will investigate that. Thanks. 

Its a full blow map that is actually too large for the screen. I have a bunch of locations on it and I need rotations to connect the locations with "streets".

The map is "scrollable". I implemented that with ui elements by setting clipping on the "view box", making the map image its child (most of the map gets clipped off) and just play with its position in reaction to player input. The locations, I implemented as glorified stylized radio buttons. Everything was beautiful and then, when the time came to connect the locations with streets (I planned to use stretching/tiling and rotation), my world turned upside down.

I will investigate "printing" the map with the generated streets on as an image as you suggest (well, I wanted to highlight some streets a times, but I can always regenerate the image). It is that or I re-implement the map and surrounding menu elements as non-ui elements, let other ui menus pop on top (that seems to be the default) and design everything else around the fact that non-ui element can never appear on top of menus.

**Reply by Magnitus-** [reply]
After looking more into it and adapting the technique in the example to my use-case, it works! Amazing! Thanks :).

That's a pretty powerful capability. I think I'll be able to overcome the bulk of my ui/non-ui incompatibility problems with this.
